src/VisualStudio/CSharp/Test/DocumentOutline/DocumentOutlineTests.cs

```diff
@@ -0,0 +1,133 @@
+                CheckSortedByName(documentSymbolItem.Children);
+
+            static void CheckSortedByName(ImmutableArray<DocumentSymbolUIItem> sortedByName)
+            {
+                for (var i = 0; i < sortedByName.Length - 1; i++)
+                    Assert.True(StringComparer.Ordinal.Compare(sortedByName[i].Name, sortedByName[i + 1].Name) <= 0);
+            }
+        }
+
+        [Fact]
+        public void TestSortByOrder()
+        {
+            var sortedByOrder = DocumentOutlineHelper.Sort(
+                DocumentOutlineHelper.GetDocumentSymbolData(GetDocumentSymbols()), SortOption.Location, CancellationToken.None);
+
+            CheckSortedByOrder(sortedByOrder);
+            foreach (var documentSymbolItem in sortedByOrder)
+                CheckSortedByOrder(documentSymbolItem.Children);
+
+            static void CheckSortedByOrder(ImmutableArray<DocumentSymbolUIItem> sortedByOrder)
+            {
+                for (var i = 0; i < sortedByOrder.Length - 1; i++)
+                {
+                    Assert.True(sortedByOrder[i].StartPosition.Line <= sortedByOrder[i + 1].StartPosition.Line);
+                    if (sortedByOrder[i].StartPosition.Line == sortedByOrder[i + 1].StartPosition.Line)
+                        Assert.True(sortedByOrder[i].StartPosition.Character < sortedByOrder[i + 1].StartPosition.Character);
+                }
+            }
+        }
+
+        [Fact]
+        public void TestSortByType()
+        {
+            var sortedByType = DocumentOutlineHelper.Sort(
+                DocumentOutlineHelper.GetDocumentSymbolData(GetDocumentSymbols()), SortOption.Type, CancellationToken.None);
+
+            CheckSortedByType(sortedByType);
+            foreach (var documentSymbolItem in sortedByType)
+                CheckSortedByType(documentSymbolItem.Children);
+
+            static void CheckSortedByType(ImmutableArray<DocumentSymbolUIItem> sortedByType)
+            {
+                for (var i = 0; i < sortedByType.Length - 1; i++)
+                {
+                    Assert.True(sortedByType[i].SymbolKind <= sortedByType[i + 1].SymbolKind);
+                    if (sortedByType[i].SymbolKind == sortedByType[i + 1].SymbolKind)
+                    {
+                        Assert.True(StringComparer.Ordinal.Compare(sortedByType[i].Name, sortedByType[i + 1].Name) <= 0);
+                    }
+                }
+            }
+        }
+
+        [Fact]
+        public void TestSearch()
+        {
+            var documentSymbolModels = DocumentOutlineHelper.GetDocumentSymbolData(GetDocumentSymbols());
+
+            Assert.Equal(2, DocumentOutlineHelper.Search(documentSymbolModels, "Class", CancellationToken.None).Length);
+            Assert.Equal(1, DocumentOutlineHelper.Search(documentSymbolModels, "Field", CancellationToken.None).Length);
+            Assert.Empty(DocumentOutlineHelper.Search(documentSymbolModels, "Banana", CancellationToken.None));
+        }*/
```

Comment:
whereforarethoutests.

Reply:
Needless to say the tests are a work in progress 😄